src/Aspire.Hosting/Dcp/Model/ModelCommon.cs

```diff
@@ -25,6 +25,8 @@ internal abstract class CustomResource : KubernetesObject, IMetadata<V1ObjectMet
     public const string OtelServiceNameAnnotation = "otel-service-name";
     public const string OtelServiceInstanceIdAnnotation = "otel-service-instance-id";
     public const string ResourceStateAnnotation = "resource-state";
+    public const string ResourceReplicaCount = "resource-replica-count";
```

Comment:
Can we achieve this without the custom DCP annotation? If you can access the IResource you should be able to validate that here.

Reply:
Karol and I talked about this. It could come from the model. But we agreed that it was kind of useful to have this info directly on a DCP resource, and it's similar to true replica sets which has this info on the DCP resource.

Will leave as annotations for now, but it is internal logic so can be changed in the future if desired.